base/multi.jl

```diff
@@ -577,20 +577,34 @@ function wait_empty(rv::RemoteValue)
 end
 
 ## core messages: do, call, fetch, wait, ref, put! ##
+type RemoteException <: Exception
+    pid::Int
+    err::Exception
+    bt::AbstractString
+end
+
+function show(io::IO, ex::RemoteException)
+    print("RemoteException on worker ", ex.pid, ": ", ex.err, "\nRemote backtrace:\n", ex.bt, "\n")
+end
 
-function run_work_thunk(thunk)
+function run_work_thunk(thunk, print_error)
     local result
     try
         result = thunk()
     catch err
-        print(STDERR, "exception on ", myid(), ": ")
-        display_error(err,catch_backtrace())
-        result = err
+        bt = catch_backtrace()
+        if print_error
+            print(STDERR, "exception on ", myid(), ": ")
+            display_error(err, bt)
+        end
+        io=IOBuffer()
+        show_backtrace(io, bt)
```

Comment:
I think profile code takes samples from the backtraces and captures only information required for profiling. The backtrace object needs to be stringified before sending to another process since the stack pointers are only valid for the local process, no?